internal/rootcoord/timeticksync.go

```diff
@@ -111,8 +111,15 @@ func (c *chanTsMsg) getTimetick(channelName string) typeutil.Timestamp {
 }
 
 func newTimeTickSync(ctx context.Context, sourceID int64, factory msgstream.Factory, chanMap map[typeutil.UniqueID][]string) *timetickSync {
+	// if the old channels number used by the user is greater than the set default value currently
+	// keep the old channels
+	defaultChanNum := Params.RootCoordCfg.DmlChannelNum.GetAsInt64()
+	if int64(len(chanMap)) > defaultChanNum {
+		defaultChanNum = int64(len(chanMap))
+	}
```

Comment:
`len(chanMap)` cannot be used as channel number. Length is actually collection number.

Reply:
Also, can we use the exceeded channel numbers here?